### PR TITLE
Feature/improve support for draft transactions

### PR DESCRIPTION
**Pull request by dtslvr**
draft-transaction

timeline-with-draft-transaction

### CODE REVIEW

File: apps/client/src/app/components/investment-chart/investment-chart.component.ts [hunk 1]
```diff
@@ -154,8 +154,8 @@ export class InvestmentChartComponent implements OnChanges, OnDestroy, OnInit {
     }
   }
 
-  private isInFuture(aContext: any, aValue: any) {
-    return isAfter(new Date(aContext?.p0?.parsed?.x), new Date())
+  private isInFuture<T>(aContext: any, aValue: T) {
```
Comment (vzickner) [inline apps/client/src/app/components/investment-chart/investment-chart.component.ts:157]: ❤️

File: apps/api/src/app/portfolio/portfolio.service.ts [hunk 2]
```diff
@@ -750,16 +764,28 @@ export class PortfolioService {
...
+    userId: string;
+  }) {
+    const where: Prisma.OrderWhereInput = { userId };
+
+    if (includeDrafts === false) {
+      where.isDraft = false;
+    }
+
     return this.orderService.orders({
+      where,
```
Comment (vzickner) [inline apps/api/src/app/portfolio/portfolio.service.ts:781]: In my opinion the entire query shouldn't be in this service

**Review by vzickner — COMMENTED** [review]
Looks good, but I think the order querying part should be inside the order service.

File: apps/api/src/app/portfolio/portfolio.service.ts (hunk 2 above)
Comment (dtslvr) [reply apps/api/src/app/portfolio/portfolio.service.ts]: Okay, I have moved `getOrders()` to the order service.

**Pull request merged**